### PR TITLE
EXPERIMENTAL BRANCH -- EXPLOSIVELY BROKEN THINGS

### PR DESCRIPTION
**Pull request by Confuseh**
minglee

### CODE REVIEW

File: src/pattyircmessage.cpp [hunk 1]
```diff
@@ -0,0 +1,160 @@
+#include "pattyircmessage.h"
+
+#define URL_REGEX "((?:https?|ftp)://\\S+)"
+
+PattyIrcMessage::url_regex(URL_REGEX);
```
Comment (pajlada) [inline src/pattyircmessage.cpp]:
``` c++
QRegularExpression url_regex = QRegularExpression(URL_REGEX);
```

File: src/pattyircmessage.h [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    // Flags parsed from IRCv3 tags
+    bool moderator;
+    bool staff;
+    bool broadcaster;
+    bool bot;
+
+    static PattyIrcMessage* fromMessage(IrcPrivateMessage* message);
+public:
+    static EmoteManager emote_manager;
+    static bool variantByIndex;
```
Comment (pajlada) [inline src/pattyircmessage.h]: remove this line

**Comment by Confuseh** [comment]
I'm planning to add some more things, so don't merge this one yet

**Comment by Confuseh** [comment]
idk whether i will add anything more on this
edit: only issues are that uncached emotes break the auto scroll after they appear on browser, ~~also it can lag out pretty heavily after scrolling at bottom if theres a lot more messages than 150, i don't know why...~~
also it spams the shit outta the console because sethtml and all emotes are re-initialised

**Comment by Confuseh** [comment]
also message looks weird when mentioned with limitting
also gif restarts with limitting

all cuz of sethtml but I don't know other way pajaW

**Pull request merged**